src/legacy/core_plugins/kibana/public/discover/np_ready/angular/discover.html

```diff
@@ -7,17 +7,28 @@ <h1 class="euiScreenReaderOnly">{{screenTitle}}</h1>
     config="topNavMenu"
     screen-title="screenTitle"
     show-search-bar="true"
+    show-filter-bar="true"
+    show-save-query="showSaveQuery"
     show-date-picker="enableTimeRangeSelector"
     index-patterns="[indexPattern]"
-
     query="state.query"
+
+    filters="filters"
+    on-filters-updated="onFiltersUpdated"
+
     on-query-submit="updateQuery"
-	
     show-save-query="showSaveQuery"
-    saved-query-id="state.savedQuery"
-    on-saved-query-id-change="updateSavedQueryId"
+    saved-query="savedQuery"
+    date-range-from="timeRangeObj.from"
+    date-range-to="timeRangeObj.to"
+    is-refresh-paused="refreshInterval.pause"
+    refresh-interval="refreshInterval.value"
+    on-refresh-change="onRefreshChange"
+
+    on-saved="onSavedQuerySaved"
+    on-saved-query-updated="onSavedQueryUpdated"
+    on-clear-saved-query="onSavedQueryCleared"
 
-    use-default-behaviors="true"
```

Comment:
Was there a necessity to add all of handlers? How it was working previously?

Reply:
I'm syncing with @lizozom to keep `use-default-behaviors`, but there are some issues with that we will have a look at.

Reply:
Update here: Will continue here once #56128 is merged, then it will be safe to use `use-default-behaviors`

Reply:
https://github.com/elastic/kibana/pull/56128 was merged
your pr will a slightly conflict because of changes in data helpers for state syncing.
Suggest to follow dashboard changes to get a better idea how exactly to use new apis
https://github.com/elastic/kibana/pull/56128/files#diff-47ad20cc6caa58a3e9c204106013514f

Reply:
Hey @Dosant !
Does it mean I have to follow new api in my pr https://github.com/elastic/kibana/pull/58352 ?

Reply:
@sulemanof, yes, please. It shouldn't be a big change.

Reply:
works now with `use-default-behaviors` 🎉